### PR TITLE
Implement action CopyToClipboardHTML

### PR DESCRIPTION
**Pull request by theoniko**
## Description

Describe your changes in detail

```markdown
Upon pressing ctrl h in the keyboard, action  CopyToClipboardHTML  is triggered which will copy to clipboard the selected text in html format.
```

## Motivation and Context

Why is this change required? What problem does it solve?

```markdown

```

## How Has This Been Tested?

- [ ] Please describe how you tested your changes.
- [ ] see how your change affects other areas of the code, etc.

## Checklist:

Go over all the following points, and put an `x` in all the boxes that apply.

If you're unsure about any of these, don't hesitate to ask. We're here to help!

- [x] I have read the [**`CONTRIBUTING`**](https://github.com/contour-terminal/contour/blob/master/CONTRIBUTING.md) document in my spoken language, and understand the terms
- [ ] I have updated (or added) the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] I have gone through all the steps, and have thoroughly read the instructions

### CODE REVIEW

**Review by christianparpart — COMMENTED** [review]
Looking back at #85, I think we need to realize this a little bit differently.

We need to add a new action, called `CopyToClipboardHTML`, and that action handler must then be implemented in `TerminalSession` (right next to the one for `CopyToClipboard`) to place an the HTML version of the text to the clipboard rather than simply the text.

Your idea with `on_mouse_select_output_type` is an interesting one, but I think users should make an explicit choice (case by case) in what format they want to copy the selected area.

**Comment by christianparpart** [comment]
An inspiration beginning would probably be something like this:

```diff
diff --git a/src/contour/Actions.h b/src/contour/Actions.h
index 18a3c815..6f86ab5a 100644
--- a/src/contour/Actions.h
+++ b/src/contour/Actions.h
@@ -22,12 +22,28 @@
 namespace contour::actions
 {
 
+// Defines the format to use when extracting a selection range from the terminal.
+enum class CopyFormat
+{
+    // Copies purely the text (with their whitespaces, and newlines, but no formatting).
+    Text,
+
+    // Copies the selection in HTML format.
+    HTML,
+
+    // Copies the selection in escaped VT sequence format.
+    VT,
+
+    // Copies the selection as PNG image.
+    PNG,
+};
+
 // clang-format off
 struct CancelSelection{};
 struct ChangeProfile{ std::string name; };
 struct ClearHistoryAndReset{};
 struct CopyPreviousMarkRange{};
-struct CopySelection{};
+struct CopySelection{ CopyFormat format = CopyFormat::Text; };
 struct CreateDebugDump{};
 struct DecreaseFontSize{};
 struct DecreaseOpacity{};
```

Then have the config loader (Config.cpp) also fill this value when parsing the YAML and making use of this enum in the implementation when text has to be extracted in the action handler.

**Comment by christianparpart** [comment]
@theoniko Would you mind rebasing your PR on top of latest master?

```sh
git fetch && git rebase origin/master
```

This will bring in also the latest prep-work #943.

**Comment by theoniko** [comment]
> @theoniko Would you mind rebasing your PR on top of latest master?
> 
> ```shell
> git fetch && git rebase origin/master
> ```
> 
> This will bring in also the latest prep-work #943.

Done

File: src/vtbackend/Terminal.cpp [hunk 1]
```diff
@@ -1139,6 +1139,44 @@ string Terminal::extractSelectionText() const
...
+    switch(cellFlags)
+    {
+        case CellFlags::Bold: result.append("font-weight: bold;\n"); break;
+        case CellFlags::Italic: result.append("font-style: italic;\n"); break;
+        case CellFlags::Underline: result.append("text-decoration: underline;\n"); break;
+        case CellFlags::DottedUnderline: result.append("text-decoration: underline dotted;\n"); break;
+        default: 
+    }
+
+    return result+ "\n}\n";
```
Comment (christianparpart) [inline src/vtbackend/Terminal.cpp:1177]: This function should not be part of the `Terminal` class. Please have a look at `VTWriter`, which basically translates the grid into the VT format as well.

You will need something similar to `VTWriter` but for HTML+CSS. You could give it the name `HtmlWriter` or `HtmlBuilder`.

**Pull request closed without merging**